### PR TITLE
Issue 303 PR: Recommended migration doc change.

### PR DESCRIPTION
**Pull request by TheBolmanator**
TL;DR  I conclude removal of the Before and After example removes a source of confusion.  
Payload property has been removed from the Action interface.  I really like the "Before" and "After" concept of this migration guide however in this case, the change seems to me to be only observable via intellisense or IDE.  I may well be wrong, but I just don't see how the change is reflected in the Before and After Case provided.  I think do think that users need to understand that when a payload is appropriate it may be added, but that the Action interface will no longer enforce inclusion of a payload property.

In the future I will be more proactive with my pull requests as I hope to be an increasingly active member this and other Repos that I frequent.  (Newbie).  

Many thanks for the great work 
Steve

### CODE REVIEW

**Comment by brandonroberts** [comment]
I don't think the Before and After example should be completely removed since we're providing guidance on migration. If anything, the explanation from here https://github.com/ngrx/platform/issues/128#issuecomment-316680916 could be incorporated into the guide.

**Comment by karptonite** [comment]
Also, if you add type information, with `ofType<FooAction>`, the IDE and compiler should correctly infer whichever properties you include on the action.

**Comment by TheBolmanator** [comment]
I will take a crack at incorporating the comment from #128 into the PR.
@karptonite I do not follow your point, and would need some additional detail to attempt to capture it.

**Comment by TheBolmanator** [comment]
Now that I have read issue #128 I am afraid to say anything on the subject.  To be perfectly honest this whole thing is over my head.   I see that even removal of an optional parameter can be a source of disquiet in the community.  A possible rewrite may be as proposed below however I am not able to translate how this fits into an "AFTER" example.  For what it is worth I just find the current "AFTER" example to be very confusing.  

PR to be resubmitted, but I am equally happy to close this with no action taken.

**Comment by brandonroberts** [comment]
Don't remove the migration example. Users still need sample code to reference. If you want to add an interface for an untyped payload, then add it as additional information.

**Comment by TheBolmanator** [comment]
No Action will be taken.
